hpcgap/lib/oper.g

```diff
@@ -1276,6 +1276,106 @@ BIND_GLOBAL( "DeclareAttribute", function ( arg )
 end );
 
 
+##############################################################################
+##
+##  This code adds thread-local attributes, which are used in HPC-GAP.
+##  These are not complete replacements for normal attributes.
+##  In particular, they are not filters, so cannot be used in method selection.
+##
+##  The main problem with thread-local attributes is that only a subset of the
+##  threads may have a value for a particular mutable attribute, but filters
+##  are not stored thread-locally. Therefore thread-local attributes are not
+##  a perfect replacement for traditional attributes.
+##
+##  The reason we have thread-local attributes is for code which uses
+##  mutable attribute as a general scratch-space, like StabChainMutable, which
+##  cannot be shared between threads.
+##
+##  Thread-local attributes where introduced to make existing
+##  GAP mutable attributes safe to use in a multi-threaded way, without
```

Comment:
This makes it sound as if there are three kinds of attributes: regular ones, mutable, and thread local. But based on everything I am reading here "between the lines", perhaps it'd be better to group them as regular and mutable ones, and then mutable ones come in "global" and thread-local variants?
(Honest question, I am deliberately *not* looking at the implementation yet, in order to assess how much of the documentation comment I can understand)